python/desc/imsim/skyModel.py

```diff
@@ -0,0 +1,136 @@
+'''
+Classes to represent realistic sky models.
+Note that this extends the default classes located in
+sims_GalSimInterface/python/lsst/sims/GalSimInterface/galSimNoiseAndBackground.py
+'''
+
+from __future__ import absolute_import, division
+
+import numpy as np
+import astropy.units as u
+
+import lsst.sims.skybrightness as skybrightness
+
+import galsim
+from lsst.sims.GalSimInterface.galSimNoiseAndBackground import NoiseAndBackgroundBase
+
+
+# Code snippet from D. Kirkby.  Note the use of astropy units.
+def skyCountsPerSec(surface_brightness=21, filter_band='r',
+                    effective_area=33.212*u.m**2, pixel_size=0.2*u.arcsec):
+    # Lookup the zero point corresponding to 24 mag/arcsec**2
+    B0 = 24.
+    s0 = (dict(u=0.732, g=2.214, r=1.681, i=1.249, z=0.862, y=0.452)[filter_band] *
+          u.electron / u.s / u.m ** 2)
+    # Calculate the rate in detected electrons / second
+    dB = (surface_brightness - B0) * u.mag(1 / u.arcsec ** 2)
+    return s0 * dB.to(1 / u.arcsec ** 2) * pixel_size ** 2 * effective_area
+
+
+# Here we are defining our own class derived from NoiseAndBackgroundBase for
+# use instead of ExampleCCDNoise
+class ESOSkyModel(NoiseAndBackgroundBase):
+    """
+    This class wraps the GalSim class CCDNoise.  This derived class returns
+    a sky model based on the ESO model as implemented in
+    """
+
+    def __init__(self, obs_metadata, seed=None, addNoise=True, addBackground=True):
+        """
+        @param [in] addNoise is a boolean telling the wrapper whether or not
```

Comment:
Anyway, consistency is better. I'm not sure whether it is more consistent to be the same as the sims_interface (especially if it will change) or the rest of DESC code.  Probably rest of DESC code.  Feel free to change this now if you can do it quickly.  Otherwise we can do it later.  I think this code is probably going to be radically restructured when we make it work per pixel so I don't think leaving it now is terrible.